### DISCUSSION TITLE
Empty interactive rebase not working

### DISCUSSION THREAD

**Discussion by Alizter** [magit/magit#4866] [Support] [answered]
When doing an interactive rebase with magit, sometimes I use it to remove commits. When the todo list is empty the rebase fails. My workaround for the moment is to add a break after the kill and then continue manually. Is this a git bug or could magit also handle this case?

**Comment by kyleam** [comment, marked as answer]
> My workaround for the moment is to add a break after the kill and then continue manually

You can also insert a noop with `z` (`git-rebase-noop`) (see #2991).  Or do a hard reset if you know upfront that you don't want any of the commits. 

> Is this a git bug

It's Git's intended behavior.  If you look in the process buffer, you'll see "error: nothing to do" (because all the items of the todo list have been dropped).

> or could magit also handle this case?

In my view we shouldn't diverge from Git's behavior.

(Transferring to a discussion because this is a feature request.)
